unstructured/ingest/connector/salesforce.py

```diff
@@ -0,0 +1,339 @@
+        return dedent(campaign)
+
+    def create_eml(self, email_json: Dict[str, Any]) -> str:
+        """Recreates standard expected .eml format using template."""
+        eml = EMAIL_TEMPLATE.substitute(
+            date=formatdate(parser.parse(email_json.get("MessageDate")).timestamp()),
+            message_identifier=email_json.get("MessageIdentifier"),
+            subject=email_json.get("Subject"),
+            from_email=email_json.get("FromAddress"),
+            to_email=email_json.get("ToAddress"),
+            textbody=email_json.get("TextBody"),
+        )
+        return dedent(eml)
+
+    @BaseIngestDoc.skip_if_file_exists
+    def get_file(self):
+        """Saves individual json records locally."""
+        self._create_full_tmp_dir_path()
+        logger.debug(f"Writing file {self.record_id} - PID: {os.getpid()}")
+
+        client = self.config._get_client()
+
+        # Get record from Salesforce based on id
+        record = client.query_all(
+            f"select FIELDS(STANDARD) from {self.record_type} where Id='{self.record_id}",
+        )
+
+        try:
+            if self.record_type == "EmailMessage":
+                formatted_record = self.create_eml(record)
+            elif self.record_type == "Account":
+                formatted_record = self.create_account(record)
+            elif self.record_type == "Lead":
+                formatted_record = self.create_lead(record)
+            elif self.record_type == "Case":
+                formatted_record = self.create_case(record)
+            elif self.record_type == "Campaign":
+                formatted_record = self.create_campaign(record)
+
+            with open(self._tmp_download_file(), "w") as page_file:
+                page_file.write(formatted_record)
+
+        except Exception as e:
+            logger.error(
+                f"Error while downloading and saving file: {self.record_id}.",
+            )
+            logger.error(e)
+
+    @property
+    def filename(self):
+        """The filename of the file created from a Salesforce record"""
+        return self._tmp_download_file()
+
+
+class SalesforceConnector(ConnectorCleanupMixin, BaseConnector):
+    ingest_doc_cls: Type[SalesforceIngestDoc] = SalesforceIngestDoc
+    config: SimpleSalesforceConfig
+
+    def __init__(
+        self,
+        config: SimpleSalesforceConfig,
+        standard_config: StandardConnectorConfig,
+    ) -> None:
+        super().__init__(standard_config, config)
+
+    def initialize(self):
+        pass
+
+    @requires_dependencies(["simple_salesforce"], extras="salesforce")
+    def get_ingest_docs(self) -> List[SalesforceIngestDoc]:
+        """Get Salesforce Ids for the records.
+        Send them to next phase where each doc gets downloaded into the
+        appropriate format for partitioning.
+        """
+        from simple_salesforce.exceptions import SalesforceMalformedRequest
+
+        client = self.config._get_client()
+
+        record_ids = []
```

Comment:
The name is a bit misleading since it's a list of docs rather than IDs

Reply:
Hmm. Interesting to think about. They are SalesforceIngestDoc . But the valuable piece we are sending through is the ids of the Salesforce records. I guess it is how you interpret the word docs. I do like a variable name to be the most accurate, so thank you.  @ryannikolaidis  thoughts?

Reply:
ohh I see. at instantiation they don't necessarily contain more than ids, but that's okay; they are instances of SalesforceIngestDoc which can use the id as the instruction set to fetch the raw document, so less "doc" in terms of raw document and rather in terms of IngestDoc. maybe we call this `ingest_docs` to remove any confusion?